debezium-core/src/main/java/io/debezium/transforms/debeziumtimestampcoverter/DebeziumTimestampConverter.java

```diff
@@ -0,0 +1,698 @@
+                        + formatPatternTimestamp, e);
+            }
+        }
+        if (formatPatternDate != null && !formatPatternDate.trim().isEmpty()) {
+            try {
+                dateFormat = new SimpleDateFormat(formatPatternDate);
+                dateFormat.setTimeZone(UTC);
+            } catch (IllegalArgumentException e) {
+                throw new ConfigException("TimestampConverter requires a SimpleDateFormat-compatible pattern for string dates: "
+                        + formatPatternDate, e);
+            }
+        }
+        config = new Config(field, type, format, dateFormat, field_type, struct_field);
+    }
+
+    @Override
+    public R apply(R record) {
+        if (operatingSchema(record) == null) {
+            return applySchemaless(record);
+        } else {
+            return applyWithSchema(record);
+        }
+    }
+
+    @Override
+    public ConfigDef config() {
+        return CONFIG_DEF;
+    }
+
+    @Override
+    public void close() {
+    }
+
+    public static class Key<R extends ConnectRecord<R>> extends DebeziumTimestampConverter<R> {
+        @Override
+        protected Schema operatingSchema(R record) {
+            return record.keySchema();
+        }
+
+        @Override
+        protected Object operatingValue(R record) {
+            return record.key();
+        }
+
+        @Override
+        protected R newRecord(R record, Schema updatedSchema, Object updatedValue) {
+            return record.newRecord(record.topic(), record.kafkaPartition(), updatedSchema, updatedValue, record.valueSchema(), record.value(), record.timestamp());
+        }
+    }
+
+    public static class Value<R extends ConnectRecord<R>> extends DebeziumTimestampConverter<R> {
+        @Override
+        protected Schema operatingSchema(R record) {
+            return record.valueSchema();
+        }
+
+        @Override
+        protected Object operatingValue(R record) {
+            return record.value();
+        }
+
+        @Override
+        protected R newRecord(R record, Schema updatedSchema, Object updatedValue) {
+            return record.newRecord(record.topic(), record.kafkaPartition(), record.keySchema(), record.key(), updatedSchema, updatedValue, record.timestamp());
+        }
+    }
+
+    protected abstract Schema operatingSchema(R record);
+
+    protected abstract Object operatingValue(R record);
+
+    protected abstract R newRecord(R record, Schema updatedSchema, Object updatedValue);
+
+
+    /**
+     * Creates the new schema of the nested struct fields that matches with config.field_type.
+     * Assign optional fields to the fields that does not contain value or null value.
+     *
+     * @param field the input nested Struct field.
+     * @param value the input nested Struct field value.
+     **/
+    private SchemaBuilder NestedStructSchemaUpdate(Field field, Struct value) {
```

Comment:
Java convention is methods should always begin with a lowercase letter. And I'd suggestion naming this method something that makes sense where the method is used (e.g., `schemaForNestedStruct(...)`).